locus/src/main/java/com/birjuvachhani/locus/Configuration.kt

```diff
@@ -44,23 +50,123 @@ data class Configuration(
+ * Note that [interval] should be bigger than 0 otherwise it throws [IllegalArgumentException].
+ *
+ * @property fastestInterval is an instance of Long, Which is the fastest allowed interval of
+ * location updates. The location updates may arrive faster than the desired [interval], but will
+ * never arrive faster than specified here. The Default value is 1000L [FASTEST_INTERVAL_IN_MS].
+ * Note that [fastestInterval] must be greater than or equal to 0 otherwise it throws [IllegalArgumentException].
+ *
+ * @property expirationTime is an instance of Long, Which is the duration of this request. A location request will not
+ * receive any locations after it has expired, and will be removed shortly thereafter.
+ * The Default value is [Long.MAX_VALUE].
+ * Note that [expirationTime] must be greater than 0 otherwise throws [IllegalArgumentException]
+ *
+ * @property numUpdates is an instance of Int, Which is the number of the location updates that we
+ * want to receive after that the LocationRequest will be removed. The Default value is [Int.MAX_VALUE].
+ * Note that [numUpdates] should be bigger than 0 otherwise it throws [IllegalArgumentException].
+ *
+ * @property smallestDisplacement is an instance of Float, Which is the minimum distance required
+ * between consecutive location updates. If a derived location update is not at least the specified
+ * distance away from the previous location update delivered to the client, it will not be delivered.
+ * The Default value is 0.
+ * Note that [smallestDisplacement] should not be smaller than 0.0F otherwise it throws [IllegalArgumentException].
+ *
+ * @property maxWaitTime is an instance of Long, Which Sets the longest a location update may be
+ * delayed. This parameter controls location batching behavior. If this is set to a value at least
+ * 2x larger than the [interval], then a device may (but is not required to) save power by
+ * delivering locations in batches.
+ * The Default value is 1000L [MAX_WAIT_TIME_IN_MS].
+ * Note This setting may lead to slow location callbacks. You are advised not to set this parameter
+ * unless necessary.
+ * Note that [maxWaitTime] must be greater than or equal to 0 otherwise it throws [IllegalArgumentException].
  */
-internal fun getDefaultRequest(): LocationRequest {
-    return LocationRequest.Builder(Configuration.INTERVAL_IN_MS)
-        .setPriority(Priority.PRIORITY_HIGH_ACCURACY)
-        .setMinUpdateIntervalMillis(Configuration.FASTEST_INTERVAL_IN_MS)
-        .setMaxUpdateDelayMillis(Configuration.MAX_WAIT_TIME_IN_MS)
-        .build()
-}
+@Parcelize
+data class LocationRequestConfiguration(
+    var priority: Int = Priority.PRIORITY_BALANCED_POWER_ACCURACY,
+    var interval: Long = INTERVAL_IN_MS,
+    var fastestInterval: Long = FASTEST_INTERVAL_IN_MS,
+    var expirationTime: Long = Long.MAX_VALUE,
+    var numUpdates: Int = Int.MAX_VALUE,
+    var smallestDisplacement: Float = 0.0F,
```

Comment:
Make sure this is the default value used by LocationRequest object!

Reply:
In GMS and HMS documentation the default value is zero.